web-common/src/features/charts/render/vega-signals.ts

```diff
@@ -0,0 +1,29 @@
+/**
+ * Util methods for handling vega signals
+ */
+
+export function resolveSignalField(value: unknown, field: string) {
+  if (typeof value === "object" && value !== null) {
+    return value[field]?.[0];
+  }
+  return undefined;
+}
+
+export function resolveSignalTimeField(value: unknown) {
+  /**
+   * Time fields end with `_ts`
+   * We iterate over the keys of the object and return the first key that ends with `_ts`
+   * */
+  if (typeof value === "object" && value !== null) {
+    for (const key in value) {
+      if (key.endsWith("_ts")) {
+        const ts = resolveSignalField(value, key);
+
+        if (ts !== undefined) {
+          return new Date(ts);
+        }
+      }
+    }
+  }
+  return undefined;
```

Comment:
**Scalability Issue**: The functions 'resolveSignalField' and 'resolveSignalTimeField' use a potentially inefficient approach for accessing object properties and could lead to performance issues when handling large datasets or complex objects. <br> **Fix**: Optimize the property access by ensuring that unnecessary computations or iterations over object properties are minimized. Consider caching results or restructuring how data is accessed to improve performance. <br> **Code Suggestion**: 
 ```
 +  // Consider using a more efficient method for property access or caching results
 +  // Example modification for resolveSignalField
 +  export function resolveSignalField(value: unknown, field: string) {
 +    if (typeof value === "object" && value !== null && Object.hasOwn(value, field)) {
 +      return value[field]?.[0];
 +    }
 +    return undefined;
 +  }
 ```